File: src/main/java/quizapi/model/Question.java
package quizapi.model;


import javax.persistence.*;
import java.util.Collection;

import static javax.persistence.CascadeType.*;

@Entity
@Table(name = "questions")
public class Question {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", unique = true, nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @Column(name = "description", nullable = false)
    private String description;

    @Column(name = "difficulty", nullable = false)
    private String difficulty;

    @OneToMany(cascade = {PERSIST, REFRESH}, targetEntity = Answer.class, mappedBy = "question")
    private Collection<Answer> answers;

    public Question() {
    }

    public Question(Category category, String description, String difficulty, Collection<Answer> answers) {
        this.category = category;
        this.description = description;
        this.difficulty = difficulty;
        this.answers = answers;
    }

    public Long getId() {
        return id;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(String difficulty) {
        this.difficulty = difficulty;
    }

    public Collection<Answer> getAnswers() {
        return answers;
    }

    public void setAnswers(Collection<Answer> answers) {
        this.answers = answers;
    }
}

File: src/main/kotlin/com/shkila/quizapi/model/Question.kt
package com.shkila.quizapi.model


import javax.persistence.CascadeType.PERSIST
import javax.persistence.CascadeType.REFRESH
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "questions")
data class Question(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", unique = true, nullable = false)
        val id: Long? = null,

        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "category_id", nullable = false)
        var category: Category? = null,

        @Column(name = "description", nullable = false)
        var description: String? = null,

        @Column(name = "difficulty", nullable = false)
        var difficulty: String? = null,

        @OneToMany(cascade = [PERSIST, REFRESH], targetEntity = Answer::class, mappedBy = "question")
        var answers: Collection<Answer>? = null
)
